### PR TITLE
go: fix Xcode/CLT 8.3-related breakage.

### PR DESCRIPTION
**Pull request by DomT4**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Ref: https://github.com/golang/go/issues/19734.
Closes https://github.com/Homebrew/homebrew-core/issues/11739.

### CODE REVIEW

**Comment by DomT4** [comment]
Can't reproduce that `godep` failure. Maybe a transient issue?

**Comment by ilovezfs** [comment]
It probably needs a revision bump. godep seems to be tied to the specific go cellar path.

Also, does this need to be backported to all the go@*?

**Comment by DomT4** [comment]
Based off [this](https://github.com/golang/go/issues/19734#issuecomment-289685513) test code, the answer seems to be yes for 1.8 & 1.7. Still testing beyond that.

**Comment by DomT4** [comment]
And 1.6...And 1.5... Oh, good. Can't test 1.4, it can't handle the `go-sdl` test case:
```
imports github.com/veandco/go-sdl2: no buildable Go source files in /tmp/gopath/src/github.com/veandco/go-sdl2
```

File: Formula/godep.rb [hunk 1]
```diff
@@ -36,6 +37,7 @@ def install
       }
     EOS
     system bin/"godep", "restore"
-    assert File.exist?("src/golang.org/x/tools/README"), "Failed to find 'src/golang.org/x/tools/README!' file"
+    assert_predicate testpath/"src/golang.org/x/tools/README", :exist?,
+                    "Failed to find 'src/golang.org/x/tools/README!' file"
   end
```
Comment (ilovezfs) [inline Formula/godep.rb]: Should this be indented by another space?

Comment (DomT4) [inline Formula/godep.rb]: Ew, yes. Good spot.

**Comment by DomT4** [comment]
I've asked upstream what the plans are IRT backporting. I'd rather not have to hack on 3 branches of Golang if I can avoid it. I suspect if they extend it they won't extend it beyond fixing the 1.7 branch, but hey, I'll take one less branch for us to work out how to fix if I can.

**Comment by ilovezfs** [comment]
Yeah let's get this in and $we can worry about the others in other PR(s) submitted by $someone.

**Comment by DomT4** [comment]
Aye, this is the biggest one, not least because it is used by other formulae and is killing at least one big one (`mongodb`).

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for taking care of this :bowing_man:

**Comment by JCount** [comment]
🥇

**Comment by DomT4** [comment]
No worries at all; I made the mistake of opening the `mongodb` bug earlier and then my curiosity was piqued 😄. Upstream has confirmed for now there are no further plans for previous branches, so either we'll need to hack on those ourselves or just see how it goes/if anyone flags it here.

**Pull request closed without merging**